schemas/2018-05-01/Microsoft.BatchAI.json

```diff
@@ -0,0 +1,1975 @@
+          "type": "string",
+          "description": "The path where the Batch AI service will store stdout, stderror and execution log of the job."
+        },
+        "inputDirectories": {
+          "oneOf": [
+            {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/InputDirectory"
+              }
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "A list of input directories for the job."
+        },
+        "outputDirectories": {
+          "oneOf": [
+            {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/OutputDirectory"
+              }
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "A list of output directories for the job."
+        },
+        "environmentVariables": {
+          "oneOf": [
+            {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/EnvironmentVariable"
+              }
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "A list of user defined environment variables which will be setup for the job."
+        },
+        "secrets": {
+          "oneOf": [
+            {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/EnvironmentVariableWithSecretValue"
+              }
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "A list of user defined environment variables with secret values which will be setup for the job. Server will never report values of these variables back."
+        },
+        "constraints": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/JobBasePropertiesConstraints"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "Constraints associated with the Job."
+        }
+      },
+      "required": [
+        "cluster",
+        "nodeCount",
+        "stdOutErrPathPrefix"
+      ],
+      "description": "The properties of a Batch AI Job."
+    },
+    "JobBasePropertiesConstraints": {
+      "type": "object",
+      "properties": {
+        "maxWallClockTime": {
+          "type": "string",
+          "default": "7.00:00:00",
+          "format": "duration",
```

Comment:
>duration [](start = 21, length = 8)

the duration format is not supported by visual studio. can you remove it?

Reply:
removed